compiling `structs.h` as C: warning: ‘struct _SNETVERSIONDATA’ declared inside parameter list will not be visible outside of this definition or declaration

I think I've seen this error before, and it has to do with `_SNETVERSIONDATA` being used before declared. The PR should be simple.

```
structs.h:1331:19: warning: ‘struct _SNETVERSIONDATA’ declared inside parameter list will not be visible outside of this definition or declaration
 1331 |      const struct _SNETVERSIONDATA *,
      |                   ^~~~~~~~~~~~~~~~
```

--- a/structs.h
+++ b/structs.h
@@ -1305,6 +1305,14 @@ typedef struct _SNETPROGRAMDATA {
 	int lcid;
 } _SNETPROGRAMDATA;
 
+typedef struct _SNETVERSIONDATA {
+	int size;
+	char *versionstring;
+	char *executablefile;
+	char *originalarchivefile;
+	char *patcharchivefile;
+} _SNETVERSIONDATA;
+
 typedef struct _SNETUIDATA {
 	int size;
 	int uiflags;
@@ -1336,14 +1344,6 @@ typedef struct _SNETUIDATA {
 	    );
 	void(* changenamecallback)();
 } _SNETUIDATA;
-
-typedef struct _SNETVERSIONDATA {
-	int size;
-	char *versionstring;
-	char *executablefile;
-	char *originalarchivefile;
-	char *patcharchivefile;
-} _SNETVERSIONDATA;
 
 // TPDEF PTR FCN UCHAR SNETSPIBIND
 // TPDEF PTR FCN UCHAR SNETSPIQUERY

